app/code/Magento/Customer/Model/EmailNotification.php

```diff
@@ -404,6 +404,10 @@ public function newAccount(
 
         $customerEmailData = $this->getFullCustomerObject($customer);
 
+        if ($customer->getExtensionAttributes() && !is_null($customer->getExtensionAttributes()->getCompanyAttributes()) && $customer->getExtensionAttributes()->getCompanyAttributes()->getStatus()!=1) {
```

Comment:
Line is too long. In this case it should be one condition per line.
You don't need to use 'is_null'
getStatus() should be compared to CompanyCustomerInterface::STATUS_ACTIVE
use !== and space correctly

Reply:
Hmm That's true but the reason I suggested was, in theory, STATUS_ACTIVE could change. Also it makes it clear in the code what you are checking.

You are referring to a value that exists only in B2B but in a standard Magento core module. ->getCompanyAttributes() is not referred to in ANY non-B2B module.

I have two suggestions - 
1. Wrap the whole if statement in a 'class_exists' check - perfectly legit (but may not be allowed due to my hierarchy statement below) -
if(class_exists(\Magento\Company\Api\Data\CompanyCustomerInterface)) { }

2. Remove this change and create a Plugin in the B2B module Magento_Company on the newAccount method.

Strictly speaking as the B2B modules are a higher layer (ie they sit on top of the main core and are optional) the lower main core modules should NOT reference them. This check you are making only happens IF and ONLY IF the B2B modules are installed - so it should be the B2B modules that make the check.

Reply:
Thanks. Updated code with plugin and created a [new PR](https://github.com/magento/partners-magento2b2b/pull/586)